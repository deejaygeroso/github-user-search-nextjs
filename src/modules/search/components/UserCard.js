import React, { Fragment } from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardHeader from "@material-ui/core/CardHeader";
import Avatar from "@material-ui/core/Avatar";
import red from "@material-ui/core/colors/red";
import CardActionArea from "@material-ui/core/CardActionArea";

const styles = {
  card: {
    maxWidth: 550,
    marginTop: 5,
    textAlign: "left"
  },
  avatar: {
    backgroundColor: red[500]
  },
};

const UserCard = ({ classes, user, onClick }) => (
  <Fragment>
    <Card className={classes.card}>
      <CardActionArea onClick={onClick}>
        <CardHeader
          avatar={
            <Avatar
              src={user.avatar_url}
              aria-label="Recipe"
              className={classes.avatar}
            />
          }
          title={user.login}
          subheader={`Followers: ${user && user.followers || 0} Following: ${user && user.following || 0}`}
        />
      </CardActionArea>
    </Card>
  </Fragment>
);

UserCard.propTypes = {
  user: PropTypes.object,
  onClick: PropTypes.func,
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(UserCard);
