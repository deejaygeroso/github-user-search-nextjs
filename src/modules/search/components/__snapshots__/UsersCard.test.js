import React from "react";
import UserCard from "../UserCard";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const user = {
    login: "username",
    followers: 650,
    following: 12
  };
  const tree = renderer
    .create(<UserCard user={user} onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
