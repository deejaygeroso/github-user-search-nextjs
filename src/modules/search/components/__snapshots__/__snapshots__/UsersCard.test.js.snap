// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root-4 MuiPaper-elevation1-7 MuiPaper-rounded-5 MuiCard-root-3 UserCard-card-1"
>
  <button
    className="MuiButtonBase-root-34 MuiCardActionArea-root-31"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <div
      className="MuiCardHeader-root-37"
    >
      <div
        className="MuiCardHeader-avatar-38"
      >
        <div
          aria-label="Recipe"
          className="MuiAvatar-root-43 MuiAvatar-colorDefault-44 UserCard-avatar-2"
        />
      </div>
      <div
        className="MuiCardHeader-content-40"
      >
        <span
          className="MuiTypography-root-46 MuiTypography-body2-54 MuiCardHeader-title-41"
        >
          username
        </span>
        <span
          className="MuiTypography-root-46 MuiTypography-body2-54 MuiTypography-colorTextSecondary-79 MuiCardHeader-subheader-42"
        >
          Followers: 650 Following: 12
        </span>
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight-33"
    />
    <span
      className="MuiTouchRipple-root-82"
    />
  </button>
</div>
`;
